go/apps/dialogue/vumi_app.js

```diff
@@ -166,6 +167,27 @@ var DialogueApp = App.extend(function(self) {
     self.states.add('states:start', function() {
         return self.states.create(self.poll.start_state.uuid);
     });
+
+    self.types.httpjson = function(desc) {
+        self.http = new JsonApi(self.im);
+
+        var payload = {
+	        user: {
+		        answers: self.im.user.answers
+	        },
+	        contact: self.contact,
+            conversation_key: self.poll.conversation_key
+	    };
+
+	    return self
+	        .http.request(desc.method, desc.url, desc.method !== 'GET' ? { data: JSON.stringify(payload) } : {})
+                .then(function(request) {
+		            if(request.body){
+			            return request.body;
+		            }
```

Comment:
The promise is fulfilled with the response data, so we should rename `request` to `response`. Could explain what's we are doing here?